core/src/player.rs

```diff
@@ -1640,11 +1644,9 @@ impl Player {
         }
 
         self.update(|context| {
-            if context.is_action_script_3() {
-                run_all_phases_avm2(context);
-            } else {
-                Avm1::run_frame(context);
-            }
+            // TODO: Is this order correct?
+            run_all_phases_avm2(context);
+            Avm1::run_frame(context);
```

Comment:
no specific comment here, I'm just... amazed this works as-is at all without major breakages or regressing perf 😅 
As for the order, could we test we somehow by loading a bunch of mixed versions in a chain and observing the order? Maybe it's still recursive in a way? Like, random guess: avm2 recursion executing avm1 exec list for the avm1movieclip when reaching it?